error.go

```diff
@@ -48,5 +48,5 @@ var errorMessages = map[Error]string{
 	ErrorUnknownIndexType:       "unknown index type",
 	ErrorEmptyID:                "document ID cannot be empty",
 	ErrorIndexReadInconsistency: "index read inconsistency detected",
-	ErrorPreSearchFailed:        "index pre-search failed",
+	ErrorPreSearchFailed:        "index pre-search failed, inconsistency in results detected, due to a topology change/rebalance. Please retry the search",
```

Comment:
Drop `Please retry the search` from error message. Also it seems you're using the same error message that's used else where - which is a bad idea. Let's declare and use a new one -

`ErrorSearchInconsistency: "Topology change caused 2-phase search to fail"`